src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -66,166 +58,181 @@ public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand,
 
     @Override
     public void update() {
-        if (coverHolder.getOffsetTimer() % 20 != 0 || !isEnabled) return;
+        if (coverHolder.getOffsetTimer() % 20 != 0) return;
 
         IEnergyContainer energyContainer = coverHolder.getCapability(GregtechCapabilities.CAPABILITY_ENERGY_CONTAINER, null);
-        if (energyContainer != null && energyContainer.getEnergyCapacity() > 0) {
-            compareValue(energyContainer.getEnergyStored(), maxEU, minEU);
-            setRedstoneSignalOutput(outputAmount);
-        }
-    }
-
-    /**
-     * Compares the value to min and max, and sets the output value accordingly.
-     * <p>
-     * Behaves like an SR Latch.
-     */
-    private void compareValue(long value, long maxValue, long minValue) {
-        if (value >= maxValue) {
-            this.outputAmount = inverted ? 15 : 0;
-        } else if (value <= minValue) {
-            this.outputAmount = inverted ? 0 : 15;
+        if (energyContainer != null) {
+            if (usePercent) {
+                if (energyContainer.getEnergyCapacity() > 0) {
+                    float ratio = (float) energyContainer.getEnergyStored() / energyContainer.getEnergyCapacity();
+                    this.outputAmount = GTUtility.computeLatchedRedstoneBetweenValues(ratio * 100, this.maxValue, this.minValue, this.isInverted, this.outputAmount);
+                    // compareValue((float) energyContainer.getEnergyStored() / energyContainer.getEnergyCapacity() * 100, maxValue, minValue);
+                } else {
+                    this.outputAmount = isInverted ? 0 : 15;
+                }
+            } else {
+                this.outputAmount = GTUtility.computeLatchedRedstoneBetweenValues(energyContainer.getEnergyStored(),
+                        this.maxValue, this.minValue, this.isInverted, this.outputAmount);
+                // compareValue(energyContainer.getEnergyStored(), maxValue, minValue);
```

Comment:
The commented out code should probably be removed, assuming this method's behavior is finalized.

Reply:
I am planning to do bigger clean up on covers #1542 so I can remove it later so we don't block next hotfix with this